### PR TITLE
Log fourmolu and ormolu version that hls using

### PR DESCRIPTION
**Pull request by July541**
Since fourmolu exported its version from fourmolu-0.12, we can utilize this to log version info.

And this pr also adds the missing fourmolu-0.13 dependency...

### CODE REVIEW

File: plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs [hunk 1]
```diff
@@ -75,6 +81,11 @@ provider recorder plId ideState typ contents fp fo = ExceptT $ withIndefinitePro
             (pure . Left . PluginInternalError . T.pack . show)
              $ runExceptT $ cliHandler fileOpts
         else do
+#if MIN_VERSION_fourmolu(0,12,0)
+            logWith recorder Info $ LogBuiltinVersion $ Just Fourmolu.version
```
Comment (michaelpj) [inline plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs:85]: Is this better than `read #VERSION_fourmolu`? That should work all the time, and for all packages, right?

File: plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs [hunk 2]
```diff
@@ -170,6 +186,8 @@ instance Pretty LogEvent where
         NoConfigPath ps -> "No " <> pretty configFileName <> " found in any of:"
             <> line <> indent 2 (vsep (map (pretty . show) ps))
         StdErr t -> "Fourmolu stderr:" <> line <> indent 2 (pretty t)
+        LogBuiltinVersion v -> "Using builtin fourmolu" <> pretty (maybe "" showVersion v)
```
Comment (michaelpj) [inline plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs:189]: I think "builtin" is a bit ambiguous. Maybe "compiled in"?

**Review by michaelpj — COMMENTED** [review]
At the moment this will be fairly noisy: it's logged at Info and it's going to be logged every time there's a formatting request. It seems to me like it would be better to do it once when the plugin is set up, if we can? That might require doing a "version probe" of the CLI tool at startup, but that seems okay?

File: plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs (hunk 1 above)
Comment (July541) [reply plugins/hls-fourmolu-plugin/src/Ide/Plugin/Fourmolu.hs:85]: Thanks! I never know this before!😅

**Comment by July541** [comment]
> That might require doing a "version probe" of the CLI tool at startup, but that seems okay?

I think this is vary from #3660, I just want to add some tracing to keep which formatter is using, hence we can locate the problem like https://github.com/haskell/vscode-haskell/issues/920 easily.

How about downgrading this log to debug level?

**Review by michaelpj — APPROVED** [review]
I'm fine with this. I'd love to do it more systematically, but this is still an improvement!

**Comment by July541** [comment]
Also log ormolu version.

**Review by fendor — CHANGES_REQUESTED** [review]
Please update the test cases as well! 

In particular, since the plugins now use the logger infrastructure, change `mkPluginTestDescriptor'` to `mkPluginTestDescriptor` for both plugins.

**Review by fendor — APPROVED** [review]
LGTM

**Pull request merged**